### PR TITLE
add pthreads4w/3.0.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **pthreads4w/3.0.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

supersedes #461
this is losely based on https://github.com/bincrafters/conan-pthreads4w, but uses configure/make as described in documentation, instead of custom cmake file

### CODE REVIEW

File: recipes/pthreads4w/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class Pthreads4WConan(ConanFile):
+    name = "pthreads4w"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/pthreads4w/"
+    description = "POSIX Threads for Windows"
+    license = "LGPL-2.0"
```
Comment (uilianries) [inline recipes/pthreads4w/all/conanfile.py]:
```suggestion
    license = "Apache-2.0"
```

According the file LICENSE in the project, it's Apache 2.0

Comment (ericLemanissier) [reply recipes/pthreads4w/all/conanfile.py]: indeed, it's apache since v3.0.0. thanks !

**Comment by madebr** [comment]
@ericLemanissier Please consider bincrafters/conan-center-index#2

File: recipes/pthreads4w/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    autotools.make(target="install-implib-default")
+                else:
+                    autotools.make(target="install-lib-default")
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if not self.options.shared:
+            self.cpp_info.defines = ["__PTW32_STATIC_LIB"]
+        self.cpp_info.defines.append(
+            '__PTW32_CLEANUP_CXX' if self.options.exception_scheme == "CPP" else\
```
Comment (SSE4) [inline recipes/pthreads4w/all/conanfile.py]: might be hard to read, but I am personally okay with that

**Pull request merged**